### PR TITLE
Misc minor (doc) fixes

### PR DESCRIPTION
**Pull request by rudis**
Minor doc fixes for a few issues I noticed while reading the docs. The first one changes git urls to always use https, not only in docs but also in a few scripts.

I'm not sure if "doc: tut2: simplify debug macro example" is entirely correct as I have no experience with nim's macros. But the result seems to work fine.

### CODE REVIEW

File: doc/tut2.rst [hunk 1]
```diff
@@ -675,16 +675,16 @@ variable number of arguments:
   macro debug(n: varargs[expr]): stmt =
     # `n` is a Nim AST that contains a list of expressions;
     # this macro returns a list of statements:
-    result = newNimNode(nnkStmtList, n)
+    result = newNimNode(nnkStmtList)
```
Comment (Araq) [inline doc/tut2.rst]: Don't do this, the ``n``  here means ``result`` has proper line information.

**Comment by Araq** [comment]
Whats wrong with git Urls?

File: doc/tut2.rst (hunk 1 above)
Comment (rudis) [inline doc/tut2.rst]: Done. I added a comment why the n is passed

**Comment by rudis** [comment]
As mentioned in the commit message git:// urls are not encrypted allowing anybody in between the client and the server to manipulate the data. https:// prevents this and is generally the recommended way to clone git repos (besides ssh access). In addition git:// uses a non-standard port which might not be accessible behind restrictive firewalls.

File: doc/tut2.rst [hunk 2]
```diff
@@ -674,17 +674,18 @@ variable number of arguments:
 
   macro debug(n: varargs[expr]): stmt =
     # `n` is a Nim AST that contains a list of expressions;
-    # this macro returns a list of statements:
+    # this macro returns a list of statements (n is passed for proper line
+    # information):
     result = newNimNode(nnkStmtList, n)
     # iterate over any argument that is passed to this macro:
-    for i in 0..n.len-1:
+    for x in n:
```
Comment (dom96) [inline doc/tut2.rst]: Might be best to give this a better name, for example ``node``.

File: lib/pure/os.nim [hunk 3]
```diff
@@ -1070,8 +1070,8 @@ proc createHardlink*(src, dest: string) =
   ## Create a hard link at `dest` which points to the item specified
   ## by `src`.
   ##
-  ## **Warning**: Most OS's restrict the creation of hard links to
-  ## root users (administrators) .
+  ## **Warning**: Some OS's restrict the creation of hard links to
```
Comment (dom96) [inline lib/pure/os.nim]: Why the change from "Most" to "Some"? The former is most appropriate, is it not?

Comment (rudis) [inline lib/pure/os.nim]: No, it is not. As the commit message says there are no systems which require a root user for hard links. Dropping the complete warning would be fine too.

File: doc/tut2.rst (hunk 2 above)
Comment (rudis) [inline doc/tut2.rst]: Just wanted a minimal patch to improve the 0..n.len-1 parameter. n looks fine to me. It's usage is also quite obvious.
